sip-servlets-client/src/main/java/org/mobicents/javax/servlet/sip/dns/DNSResolver.java

```diff
@@ -60,7 +61,7 @@
 	 * @param uri the uri used to find the corresponding SipURI
 	 * @return the SipURI found through ENUM methods or the uri itself if the uri is already a SipURI without a user=phone param
 	 */
-	SipURI getSipURI(URI uri);
+	SipURI getSipURI(URI uri, SipSession sipSession);
```

Comment:
ummm, using the system to commumnicate those values is not ideal, but I see where you coming from...

Threorically, all managed threads have a threadlocal instance with access to current sessions, so if we manage to recover that threadlocal instance, you may query the context everywhere. Let me think how to get that instance ok?

 https://github.com/RestComm/sip-servlets/blob/master/sip-servlets-impl/src/main/java/org/mobicents/servlet/sip/core/session/SipApplicationSessionCreationThreadLocal.java